### DISCUSSION TITLE
How to add a shortcut in watch face to link calendar?

### DISCUSSION THREAD

**Discussion by fzhying** [zepp-health/zeppos-forum#161] [Q&A] [answered]
How to use hmApp.startApp start calendar? What are the appid and the url of calendar in Zepp OS?

**Comment by zdunex25** [comment, marked as answer]
_hmApp.startApp({ appid: 1, url: '**ScheduleCalScreen**', native: true })_

**Reply by fzhying** [reply]
Thanks！

**Reply by fzhying** [reply]
If possible, please public all the appids and the urls of built-in applications in Zepp OS.

**Comment by htoooth** [comment]
内置的应用都是在内部使用的，并不会公开支持。这块不做一直能够使用的保证。另外，我们计划对内置的应用也使用 appId。到时候方便大家跳转。

The built-in applications are for internal use only and are not publicly supported. We cannot guarantee their continuous availability. In addition, we plan to use an appId for the built-in applications as well, which will make it easier for everyone to navigate to them.

**Reply by zdunex25** [reply]
Would be really handy to add more **data_types**, so we could simply use **IMG_CLICK**.

**Reply by htoooth** [reply]
@zdunex25 /gpt 
这个功能我知道，是表盘组件的跳转功能，目前这个跳转功能可能会不变。我们规划功能会在表盘和应用都能使用。

I am aware of this feature, which is the jump function of the watch face component. Currently, this jump function may remain unchanged. We are planning to make this feature available for both watch faces and applications.

**Reply by fzhying** [reply]
hmApp.startApp  is available for both watch faces and applications for building hyperlinks while relying on the appid and the url of the linked application. In the watch face component, all data_types for IMG_CLICK have been introduced in [official technical document](https://docs.zepp.com/docs/1.0/watchface/api/hmUI/widget/data_type/) while they are so limited. Once the appid and the url of the linked application are identified, we could also simply build hyperlinks via hmApp.startApp. That is why I asked whether the official would make all the appids and the urls of built-in applications in Zepp OS public. However, the official has claimed that the appids and the urls of the applications in Zepp OS are not all available for public query.

> Would be really handy to add more **data_types**, so we could simply use **IMG_CLICK**.

**Reply by htoooth** [reply]
@fzhying
1. 内置的官方原生应用的 appid 和 url，我们会有计划提供。
2. 第三方开发者应用之间相互跳转，这块第三方应用的的 apid 和 url 这块，我们不会提供。

-----

1. We have plans to provide the appid and url for the built-in official native applications.
2. We will not provide the apid and url for third-party developer applications that allow for mutual jumping between them.

**Reply by zdunex25** [reply]
@htoooth is there any chance to get appid/url or at least data_type for Membership Cards? We cannot add them to widgets or shortcuts, the only way to open is through apps list.
![image](https://github.com/zepp-health/zeppos-forum/assets/1674725/d84f6340-d7b8-462a-afe2-b40f6c10bf89)

**Reply by htoooth** [reply]
@zdunex25 This feature is not available yet.

**Comment by zdunex25** [comment]
> 1. We have plans to provide the appid and url for the built-in official native applications.

@htoooth any news? I would like to add shortcut to my app (os 2.1) that will take user to sleep page, but startApp is only available for os 1.0

**Reply by zdunex25** [reply]
@silver-zepp maybe you will help I saw your live stream on youtube about os 3.0 and i believe you have the knowledge

**Reply by silver-zepp** [reply]
The `startApp` was replaced with [lauchApp](https://docs.zepp.com/docs/v2/reference/device-app-api/newAPI/router/launchApp) and the sleep page that you're looking for I think that's `"Sleep_HomeScreen"` or the SLEEP constant from [here](https://docs.zepp.com/docs/reference/device-app-api/newAPI/router/checkSystemApp). But I'm not sure it can open built-in apps. @htoooth is working on this API and he should be available from Monday next week @zdunex25

**Reply by zdunex25** [reply]
Thanks, playing with **launchApp** I figured it out:

![image](https://github.com/zepp-health/zeppos-forum/assets/1674725/a0d80cd4-2af5-4528-97a7-c9117f7ebd06)

**Reply by underGhost** [reply]
Does anyone know if it's possible to deep link? I want to go to SYSTEM_APP_SETTING -> Display
I see some watch faces can. Also want to go to SYSTEM_APP_HR -> details (calories)

**Reply by zdunex25** [reply]
> Does anyone know if it's possible to deep link? I want to go to SYSTEM_APP_SETTING -> Display I see some watch faces can. Also want to go to SYSTEM_APP_HR -> details (calories)

1) hmApp.startApp({url: 'Settings_displayBrightScreen', native: true });
2) hmApp.startApp({url: 'activityAppScreen', native: true });

**Reply by underGhost** [reply]
Thanks @zdunex25 , is there an anchor tag to get it to scroll down to the `details` section automatically? I noticed some apps take you straight to the details section on `activityAppScreen`.

**Reply by zdunex25** [reply]
@underGhost I only know: hmApp.startApp({url: 'activityWeekShowScreen', native: true }); so try and see if it suits you.

**Reply by underGhost** [reply]
@zdunex25 Thanks for the suggestion. Ideally i'd like to still point to `activityAppScreen` screen, just a different section of data, without the need for the user to scroll.

@htoooth Is there a list of [system app](https://docs.zepp.com/docs/reference/device-app-api/newAPI/router/launchApp/#system-app-id-constants) -> sub pages? Or even sub sections within a page? Would be really helpful. I see other watch faces can navigate places that aren't captured in the documentation. (For instance the above `activityAppScreen` -> details.

**Reply by zdunex25** [reply]
> Thanks @zdunex25 , is there an anchor tag to get it to scroll down to the `details` section automatically? I noticed some apps take you straight to the details section on `activityAppScreen`.

    There is no official list that's for sure.
